lib/cartopy/feature.py

```diff
@@ -123,6 +123,90 @@ def intersecting_geometries(self, extent):
             return self.geometries()
 
 
+class Scaler(object):
+    """
+    General object for handling the scale of the geometries used in a Feature.
+    """
+    def __init__(self, scale):
+        self._scale = scale
+
+    @property
+    def scale(self):
+        return self._scale
+
+    def scale_from_extent(self, extent):
+        """
+        Given an extent, update the scale.
+
+        Parameters
+        ----------
+        extent
+            The extent over which we should choose a scale. The
+            coordinate system of the extent should be constant, and at the
+            same scale as the scales argument in the constructor.
+
+        """
+        # Note: Implementation does nothing. For subclasses to specialise.
+        return self._scale
+
+
+class AdaptiveScaler(Scaler):
+    """
+    Object for changing scales of geometries based on extent of axes used to
+    plot geometries.
+    """
+    def __init__(self, default_scale, limits):
+        """
+        Parameters
+        ----------
+        default_scale
+            Coursest scale used as default when plot is at maximum extent.
+
+        limits
+            Scale-extent pairs at which scale of geometries change. Must be a
+            tuple of tuples ordered from coursest to finest scales. Limit
```

Comment:
coarse with an a?

Reply:
Degrees is not necessarily correct here, this implementation works with non PC coord systems.